services/app-api/src/postgres/contractAndRates/parseRateWithHistory.ts

```diff
@@ -165,42 +166,58 @@ function rateWithHistoryToDomainModel(
                 )
             }
 
-            // if it's from before this rate was submitted, it's there at the beginning.
+            // Finding the single earliest contract rev submitted right after rate rev submission date time.
```

Comment:
This comment confuses me. Isn't this code finding multiple revs not just a single contract rev? Seems like we adding any contract rev that was submitted after rate rev to `initialEntry.contractRevs` with an array push.

Also why are keeping all this commented old code along? I think we can delete - its in version history.

Reply:
This is building the rate history using the `rate.revisions` in the type `RateType`. 
- Since `rate.revisions` are all submitted rates, we need to attach the single contract revision that was submitted along with this rate to preserve the same submission data at that time.
-  We don't want future submission revisions of the same contract attached to it otherwise, this data would have future edits trickle down the revision history.

Let me know if that makes sense.